README.md

```diff
@@ -706,6 +706,21 @@ Similarly, defining tables out-of-order is discouraged as well.
 [b]
 ```
 
+Dotted keys imply the creation of tables. Once created, the `[table]` form can
+only be used to define sub-tables within such tables.
```

Comment:
These two sentence are correct, but their terseness makes them hard to understand. I would be somewhat more verbose, maybe as follows:

> By using dotted keys, you implicitly define everything to the left of each dot as a table. Since tables cannot be defined more than once, redefining such tables using a `[table]` header is not allowed. Using dotted keys to redefine tables already defined in `[table]` form is likewise not allowed. The `[table]` form can, however, be used to define sub-tables within tables defined in dotted-key form.